debezium-core/src/main/java/io/debezium/relational/topic/ByDatabasePlatformTopicMapper.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.relational.topic;
+
+import org.apache.kafka.connect.data.Schema;
+import org.apache.kafka.connect.data.SchemaBuilder;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * @author David Leibovic
+ * @see this doc for useful terminology definitions: https://docs.google.com/document/d/12KZDiv5uUvlB8jnJQJ8KK7PR6whPoWUQEWpX-Ly8n3A/edit#
+ */
+public class ByDatabasePlatformTopicMapper extends TopicMapper {
```

Comment:
So would we allow more than one `TopicMapper` per connector? If so, then couldn't we just reserve the `topic.mapper.*` prefix for any custom topic mapper configuration properties? That way we don't have to validate them, either. The code could get the mapper-related properties with `config.subset("topic.mapper.", false)` and pass that to the initializer method after instantiation.

Any `TopicMapper` implementations that we provide can use custom properties as need, so yes `topic.mapper.sharding.regex` would be valid.